webextension/background.js

```diff
@@ -1,3 +1,174 @@
+const assignManager = {
+  closeableWindows: new Set([
```

Comment:
Using uppercase here makes me cry a little (especially when it's not a const too), much like Hungarian notation does but I will not argue ;)